src/main/java/com/faforever/client/chat/ChannelTabController.java

```diff
@@ -235,6 +252,35 @@ public void setChannel(Channel channel) {
     searchFieldContainer.visibleProperty().bind(searchField.visibleProperty());
     closeSearchFieldButton.visibleProperty().bind(searchField.visibleProperty());
     addSearchFieldListener();
+    topicPane.managedProperty().bind(topicPane.visibleProperty());
+    updateChannelTopic();
+    JavaFxUtil.addListener(channel.topicProperty(), new WeakInvalidationListener(channelTopicListener));
+  }
+
+  private void updateChannelTopic() {
+    boolean hasTopic = channel.getTopic() != null;
+    topicPane.setVisible(hasTopic);
+    if (!hasTopic) {
+      return;
+    }
+    String topic = channel.getTopic() == null ? "" : channel.getTopic();
+    Arrays.stream(topic.split("\\s"))
+        .forEach(word -> {
+          if (URL_REGEX_PATTERN.matcher(word).matches()) {
+            Hyperlink link = new Hyperlink(word);
+            link.setOnAction(event -> platformService.showDocument(word));
+            topicText.getChildren().add(link);
+          } else {
+            topicText.getChildren().add(new Label(word + " "));
+          }
+        });
+  }
+
+  @Subscribe
+  public void onChannelTopicChangedEvent(ChannelTopicChangedEvent channelTopicChangedEvent) {
```

Comment:
What is this needed for? I see no code that actually sends these topic events, and you should already get all topic changes through the `Channel` objects themselves (which in turn get their update from IRC `TopicEvent`), right?